odoo/upgrade_code/18.1-02-import-class.py

```diff
@@ -0,0 +1,99 @@
+
+
+def upgrade(file_manager):
+    nb = sum(1 for file in file_manager if file.path.name.endswith('.py'))
+    i = 0
+    file_manager.print_progress(i, nb)
+
+    class_to_import = defaultdict(dict)
+
+    for file in file_manager:
+        if not file.path.name.endswith('.py'):
+            continue
+
+        i += 1
+        file_manager.print_progress(i, nb)
+
+        # Import all odoo classes which have model name matching class name
+        file_classes = set(regex_class_name.findall(file.content)) - set(regex_name.findall(file.content)) - set(regex_inherit.findall(file.content))
+        if not file_classes:
+            continue
+
+        # Updated imports so that classes are imported in their parent (Most of the time it is about 'models' folder)
+        modules = []
+        parent = file.path
+        parent_init = None
+        while parent != file.addon:
+            parent_init = file_manager.get_file(PurePath(parent.parent, Path('__init__.py')))
+            if not parent_init:
+                break
+
+            module = parent.name.split('.')[0]
+            modules.append(module)
+
+            if not (re.search(rf'from *\.\b{module}\b', parent_init.content) or
+                    re.search(rf'from *\.\b{".".join(reversed(modules))}\b', parent_init.content) or
+                    re.search(rf'from *\. +import .*\b{module}\b', parent_init.content)):
+                parent_init = None
+                break
+
+            parent = parent.parent
+
+        if not parent_init:
+            continue
+
+        module = '.'.join(reversed(modules))
+        class_to_import[parent_init][module] = file_classes
+
+    for parent_init, imported_modules in class_to_import.items():
+        modules = sorted(imported_modules.keys())
+        content = parent_init.content
+
+        imports = []
+        for module in modules:
+            file_classes = imported_modules[module]
+            class_names = {c for c in file_classes if not re.search(rf'\b{c}\b', content)}
+            if class_names:
+                # Some classes must be imported
+                imports.append(join_classes_to_string_import(module, class_names))
+        if not imports:
+            continue
+
+        import_ref = re.findall(rf'(from.* import .*)', content)[-1]
+        index = content.index(import_ref) + len(import_ref)
+        imports_str = '\n'.join(imports)
+        content = f"{content[:index]}\n{imports_str}\n{content[index+1:]}"
+        if content[-1] != "\n":
+            content += "\n"
+        parent_init.content = content
+
+
+def join_classes_to_string_import(module_name, class_names):
+    """ return to string representing the import to be done for the classes.
+        Automatically wraps lines to write lines of up to 100 characters.
+    """
+    import_str = rf'from .{module_name} import'
+    line_len = len(import_str)
+    classes_str = ""
+    multi_line = False
```

Comment:
You could just say: `multiline = sum(len(c) + 2 for c in class_names) + len(import_str) > 99`
Then the rest will be simpler. You could even put one class per line in case of multiline.